floodlight/io/statsperform.py

```diff
@@ -0,0 +1,270 @@
+
+    # create periods for segments, coded as jumps in the frame sequence
+    periods = {}
+    frame_values = dat_df["frame_count"].unique()
+
+    seg_idx = np.where(np.diff(frame_values, prepend=frame_values[0]) > 1)
+    seg_idx = np.insert(seg_idx, 0, 0)
+    seg_idx = np.append(seg_idx, len(frame_values))
+    for segment in range(len(seg_idx) - 1):
+        start = int(frame_values[seg_idx[segment]])
+        end = int(frame_values[seg_idx[segment + 1] - 1])
+        periods[segment] = (start, end)
+
+    return framerate, periods, pitch
+
+
+def _create_links_from_dat_df(
+    dat_df: pd.DataFrame, team_ids: Dict[str, float]
+) -> Dict[str, Dict[int, int]]:
+    """Creates links between player_id and column in the array from parsed dat-file.
+
+    Parameters
+    ----------
+    dat_df: pd.DataFrame
+        Containing the parsed DataFrame from the positions CSV file
+
+    Returns
+    -------
+    links: Dict[str, Dict[int, int]]
+        Dictionary with frame axis for the given segment of the form
+            `periods[segment] = (startframe, endframe)`.
+    """
+    links = {}
+    for team in team_ids:
+        links[team] = {
+            jID: xID + 1
+            for xID, jID in enumerate(
+                dat_df[dat_df["team_id"] == team_ids[team]]["jersey_no"].unique()
+            )
+        }
+    return links
+
+
+def create_links_from_dat(filepath_dat: Union[str, Path]) -> Dict[str, Dict[int, int]]:
+    """Creates links between player_id and column in the array from dat-file.
+
+    Parameters
+    ----------
+    filepath_dat: str or Path
+        XML File where the Position data in Statsperform format is saved.
+
+
+    Returns
+    -------
+    links: Dict[str, Dict[str, int]]
+        Dictionary mapping the player jersey number to his position in the data array.
+    """
+    # read dat-file into pd.DataFrame
+    dat_df = pd.read_csv(str(filepath_dat))
+
+    # initialize team and ball ids
+    team_ids = {"Home": 1.0, "Away": 2.0}
+    ball_id = 4
+    # check
+    assert [ID in team_ids or ID == ball_id for ID in dat_df["team_id"].unique()]
+
+    return _create_links_from_dat_df(dat_df, team_ids)
+
+
+def read_positions(
+    filepath_dat: Union[str, Path],
+    links: Dict[str, Dict[int, int]] = None,
+) -> Tuple[XY, XY, XY, XY, XY, XY, Code, Code, Pitch]:
+    """Parse StatsPerform dat file for (x,y)-data.
+
+    Read a position data CSV file that is given in the StatsPerform format into a
+    Tuple of six floodlight.core.xy.XY Objects comprising a Tuple of six
+    floodlight.core.xy.XY objects comprising respectively the position data for the both
+    halftimes of the game for the home team, away team, and the ball.
+
+    Parameters
+    ----------
+    filepath_dat: Union[str, Path]
+        XML File where the Position data in Statsperform format is saved.
+    links: Dict[str, Dict[int, int]] or None
+        Dictionary mapping the player jersey number to his position in the data array,
```

Comment:
jID to xID?

Reply:
Yes, I refactored the comments largely based on tracab.py